### PR TITLE
gha(validation): Make room for other validations

### PR DESCRIPTION
**Pull request by chadwhitacre**
- Broaden scope of action slightly to be about validation in general, not just validation against a template (makes room for rejecting entirely empty issues).
- Adds link back to logs for visibility and easier debugging.
- Picks off a few nits in the process.

### CODE REVIEW

**Comment by chadwhitacre** [comment]
CI is green: https://github.com/getsentry/.github-chadwhitacre/actions/runs/531261148.

File: .github/workflows/test-helpers/get-issue-validation-log [hunk 1]
```diff
@@ -39,7 +39,7 @@ function download-and-extract {
   actual=$(grep 'Validating issue #[0-9]*\.[^"]' log | sed -Ee 's/.*#([0-9]*)\..*/\1/')
   echo "Expected: ${expected}"
   echo "Actual:   ${actual}"
-  test $actual = $expected
+  test "$actual" = "$expected"
```
Comment (chadwhitacre) [inline .github/workflows/test-helpers/get-issue-validation-log:42]: nit: This was failing with a less-than-helpful error message if one of the args was empty.

File: .github/workflows/validate-new-issue-test.yml [hunk 2]
```diff
@@ -105,7 +105,7 @@ jobs:
           issue_number=$(
             basename $(
               gh issue create --title "Greetings, program!" \
-                --body "# Foo\nI don't follow directions."
+                --body "I don't follow directions."
```
Comment (chadwhitacre) [inline .github/workflows/validate-new-issue-test.yml:108]: Nit: The `\n` was coming through verbatim, not as a newline. [E.g.](https://github.com/getsentry/.github-chadwhitacre/issues/416).

File: .github/workflows/validate-new-issue.yml [hunk 3]
```diff
@@ -50,15 +57,18 @@ jobs:
...
+          Sorry, friend. As far as this ol' bot can tell, ${REASON}. Please [try again](https://github.com/${{ github.repository }}/issues/new/choose), if you like. (And if I'm confused, please [let us know](https://github.com/getsentry/.github/issues/new?title=template+enforcer+is+confused&body=${{ github.event.issue.html_url }}). 😬)
 
           ----
 
-          [![Did you see the memo about this?](https://user-images.githubusercontent.com/134455/104515469-e04a9c80-55c0-11eb-8e15-ffe9c0b8dd7f.gif)](https://www.youtube.com/watch?v=Fy3rjQGc6lA)"
+          [![Did you see the memo about this?](https://user-images.githubusercontent.com/134455/104515469-e04a9c80-55c0-11eb-8e15-ffe9c0b8dd7f.gif)](https://www.youtube.com/watch?v=Fy3rjQGc6lA)
```
Comment (chadwhitacre) [inline .github/workflows/validate-new-issue.yml:66]: Nit: Quote at the end is left over from the old days (i.e., yesterday) when this was formatted as JSON. cf. https://github.com/getsentry/.github/pull/36

**Review by BYK — APPROVED** [review]
Looks great, thanks @chadwhitacre!

**Comment by chadwhitacre** [comment]
Landed in #38.

**Pull request closed without merging**